File: Leet Code/Partitioning Into Minimum Number Of Deci-Binary Numbers/answer.java
class Solution {
    private int value(char num){
        if(num == '0'){
            return 1;
        }
        else{
            return num - '0';
        }
    }
    public int minPartitions(String n) {
        int sum = 0;

        for(int i=0; i<n.length(); i++){
            int val = value(n.charAt(i)) - sum;

            if(val < 0){
                sum += 0;
            }
            else{
                sum += val;
            }
        }
        return sum;
    }
}

// Approach 2
class Solution {
    public int minPartitions(String n) {
        return n.chars().max().getAsInt()-'0';
    }
}

File: Leet Code/Partitioning Into Minimum Number Of Deci-Binary Numbers/answer.cpp
class Solution {
public:
    int value(char num){
        if(num == '0'){
            return 1;
        }
        return int(num) - 48;
    }

    int minPartitions(string n) {
        int ans = 0;

        for(int i=0; i<n.length(); i++){
            
            int val = value(n[i]) - ans;
            (val < 0)? ans += 0 : ans += val;
        }

        return ans;
    }
};

// Approach 2

class Solution {
public:
    int minPartitions(string n) {
        int maxx = INT_MIN;

        for(auto x:n){
            maxx = max(x-'0',maxx);
        }

        return maxx;
    }
};
